shell/common/node_bindings.cc

```diff
@@ -158,19 +158,37 @@ void V8FatalErrorCallback(const char* location, const char* message) {
 }
 
 bool AllowWasmCodeGenerationCallback(v8::Local<v8::Context> context,
-                                     v8::Local<v8::String>) {
+                                     v8::Local<v8::String> source) {
   // If we're running with contextIsolation enabled in the renderer process,
   // fall back to Blink's logic.
   v8::Isolate* isolate = context->GetIsolate();
   if (node::Environment::GetCurrent(isolate) == nullptr) {
     if (gin_helper::Locker::IsBrowserProcess())
       return false;
     return blink::V8Initializer::WasmCodeGenerationCheckCallbackInMainThread(
-        context, v8::String::Empty(isolate));
+        context, source);
   }
 
-  return node::AllowWasmCodeGenerationCallback(context,
-                                               v8::String::Empty(isolate));
+  return node::AllowWasmCodeGenerationCallback(context, source);
+}
+
+v8::ModifyCodeGenerationFromStringsResult ModifyCodeGenerationFromStrings(
+    v8::Local<v8::Context> context,
+    v8::Local<v8::Value> source,
+    bool is_code_like) {
+  // If we're running with contextIsolation enabled in the renderer process,
```

Comment:
Not sure which part of this logic is specific to ctx isolation, looks like we ask blink in all non-browser processes?

Reply:
Does this work in the new utility processes?

Reply:
when context isolation is disabled, there is a node env for the main world, so we get the node behavior for this function.

there's no blink in the utility process and there is a node env for the context there so this works just the same as main there.